### ISSUE TITLE
win recorders dont get cpu time

### ISSUE THREAD

**Issue by sparvu** [kronometrix/recording#85]
It has been observed on some systems that kdr recorders sometimes do not run. They do not store data to disk immediately. They are up and running, as win32 services.

KDR recorders, by default run as NORMAL priority. On a busy system, it is highly possible Win32 kernel will not give CPU to them unless they operate on HIGH or REALTIME priority. 

We have noticed similar behavior on our lab system, where sometimes during peak load some recorder wont write to raw data files immediately.

**Comment by sparvu** [comment]
Most likely on some systems kdr recorders will run when the system and kernel will offer the CPU. So to enhance this part and make sure always kdr recorders will run we should set the priority of the recorders and transporter to HIGH priority instead of NORMAL.

Proposed change:

```
if (RunningAsService()) {

    # get service sample interval
    $interval = getssi($cdata);
    $loop_max = 2**32;

    $rawlog = open_file($fkrd);
    $log = open_file($fname);

    my $curentProcess;

    my $pid = Win32::Process::GetCurrentProcessID();

    if (Win32::Process::Open($curentProcess, $pid, 0)) {
        $curentProcess->SetPriorityClass(HIGH_PRIORITY_CLASS);
    }

    writelog ("info: started");
}


```

**Comment by sparvu** [comment]
This has been tested entire day with the following observation:

 * recorders do get more often CPU than using NORMAL priority class

 * the change does not add any additional CPU usage, which is very good

 * the overall sysrec cpu usage on a busy 75% system is 0.02 usage to 0.5

**Comment by sparvu** [comment]
fixed in 1.0.19 rc1
